src/main/util.ts

```diff
@@ -247,74 +264,101 @@ const getVideoState = (storageDir: unknown) => {
 }    
 
 /**
- *  writeMetadataFile
+ *  Writes video metadata asynchronously and returns a Promise
  */
-const writeMetadataFile = (storageDir: string, metadata: Metadata) => {
+const writeMetadataFile = async (storageDir: string, metadata: Metadata) => {
+    const file = await getNewestVideo(storageDir);
```

Comment:
I probably wrote this behaviour, but I think it's a bit crap that this function writes Metadata for the newest video. Seems like doing that will expose us to all sorts of confusion. Can we make the video to write metadata for an arg we pass in? Fine if you want to leave for me as this is again pre-existing.

Reply:
Fixed.